### PR TITLE
(Almost) final version

### PR DESCRIPTION
**Pull request by ValentinJSchmidt**
This fixes the last  big problems with the code
Counting total words should now work
More comments
Check corpus now works the same for every situation
Word Types are now counted
Code was linted ( except some parts that @derNarr wrote)
Documentation was improved

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,8 +2,100 @@ __pycache__/
...
+docs/tests/no_phoneme.pkl
 *.wav
+*.swp
+poetry.lock
+*seg
+*ges
+manually_add_mels.py
+*csv
 
+# Byte-compiled / optimized / DLL files
```
Comment (derNarr) [inline .gitignore]: From here onward it looks like a pretty exhaustive list. Do we really need to gitignore all these? I am rather favoring a "minimal" gitignore than an exhaustive one.

File: README.rst [hunk 2]
```diff
@@ -30,12 +35,102 @@ functions from top to bottom. The functions are supplied by the other files.
...
+
+    corpus/
+    ├── validated.tsv         # a file where the transcripts are stored
+    ├── clips/
+    │   └── *.mp3             # audio files (mp3)
+    └── files_not_relevant_to_this_project
+
+If you run the following command the package will align the audio files for you, and then create a pandas DataFrame with the synthesized audio and other information useful for the PAULE model,
+but only for the first 100 words that occur 4 times or more. Since you use multiprocessing, no melspectrograms are generated.:
+.. code:: bash
```
Comment (derNarr) [inline README.rst:140]: empty line missing before ``..code:: bash``?

File: .zenodo.json
Comment (derNarr) [inline .zenodo.json:1]: This file needs to be removed. IMHO

File: docs/requirements.txt [hunk 3]
```diff
@@ -4,4 +4,15 @@ notebook>=6.4.10
 seaborn>=0.11.2
 numpydoc>=1.2
 easydev>=0.9.35
-nbsphinx>=0.8.8
\ No newline at end of file
+nbsphinx>=0.8.8
+sphinx-argparse>=0.2.5
+numpy==1.26.4
```
Comment (derNarr) [inline docs/requirements.txt]: Do we still need to pin numpy to 1.26.4? I would rather have no pinned versions in here.

File: docs/source/glossary.rst [hunk 4]
```diff
@@ -16,15 +17,22 @@ Control Parameter Trajectories (cp-trajectories)
 
 Montreal Forced Aligner (MFA)
 =============================
+The Montreal Forced Aligner (MFA) is a speech processing tool that aligns speech to its transcript.
+It is developed by MontrealCorpusTools at McGill university in Montreal. You can find more information about MFA on the `MFA website <https://montreal-forced-aligner.readthedocs.io/en/latest/>`_.  
 
 
 Phonetic alphabet
 =================
```
Comment (derNarr) [inline docs/source/glossary.rst]: Remove empty lines below "underlines of headings".

File: docs/source/issues.rst [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+=================================================================
+Words that would be skipped because they occur less than 4 times in the corpus are not counted in the total amount of words in the corpus or in the amount lost words.
+The amount of words in the corpus and the amount of lost words are only estimated if a word seperation error occurs. We then simply split the transcript by spaces and count the words and divide them by a factor of 1.001 since
+some of the words in a sentence might not occur 4 times or more, so we would have skipped them anyways. Please adjust the factor for your own corpus if you have a different word distribution and/or a 
+minimum word count that is different from our standard 4 times or more.
+For other secenarios the words are counted and  we follow the word seperation done by the aligner. 
+
+Estimation of total word types
+==============================
+
```
Comment (derNarr) [inline docs/source/issues.rst]: remove empty line

File: docs/tests/test_mfa_to_sampa_dict.py [hunk 6]
```diff
@@ -17,6 +17,9 @@
 
 
 def test_mfa_to_sampa():
+    """
+    This test checks if the mfa_to_sampa_dict is working and produces a phoneme. It does not check if the phoneme is acutally accepted by VTL
```
Comment (derNarr) [inline docs/tests/test_mfa_to_sampa_dict.py]: break into two lines

File: docs/tests/test_mfa_to_sampa_dict.py [hunk 7]
```diff
@@ -61,50 +64,30 @@ def test_mfa_to_sampa():
...
-    if not os.path.exists(os.path.join(path, "temp_output", "phones")):
-        os.mkdir(path=os.path.join(path, "temp_output", "phones"))
-    else:
-        shutil.rmtree(os.path.join(path, "temp_output", "phones"))
-        os.mkdir(path=os.path.join(path, "temp_output", "phones"))
-    sf.write(
-        f"tests/clips/temp_output/phones/no_phoneme.wav", no_phoneme, no_phoneme_sr
-    )
+    """
+    This test checks if each phoneme in the mfa_to_sampa_dict is actually a phoneme and not just silent noise since VTL will not give an error if the phoneme is not a phoneme
```
Comment (derNarr) [inline docs/tests/test_mfa_to_sampa_dict.py]: Break into two lines

File: poetry.lock
Comment (derNarr) [inline poetry.lock:1]: Do we need this file in the repo?

**Review by derNarr — CHANGES_REQUESTED** [review]
Please change some of the small things I commented.

File: poetry.lock
Comment (ValentinJSchmidt) [reply poetry.lock:1]: no

**Pull request merged**